app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -227,6 +235,14 @@ def log_object_update(object_update)
     end
   end
 
+  def full_refresh_needed?
+    (Time.now.utc - last_full_refresh) > full_refresh_interval
```

Comment:
Just checking - could `last_full_refresh` ever be `nil` here? I'm thinking no but not sure.

Reply:
Good question, it is set by [`full_refresh`](https://github.com/ManageIQ/manageiq-providers-vmware/blob/70ceee49c0676322eb72e0a2d70492aee99448d5/app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb#L54) [here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/70ceee49c0676322eb72e0a2d70492aee99448d5/app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb#L24) before `full_refresh_needed?` is ever called [here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/70ceee49c0676322eb72e0a2d70492aee99448d5/app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb#L28)